packages/flutter/lib/src/painting/decoration.dart

```diff
@@ -9,22 +9,105 @@ export 'edge_insets.dart' show EdgeInsets;
 
 // This group of classes is intended for painting in cartesian coordinates.
 
+/// A description of a box decoration (a decoration applied to a [Rect]).
+///
+/// This class presents the abstract interface for all decorations.
+/// See [BoxDecoration] for a concrete example.
+///
+/// To actually paint a [Decoration], use the [createBoxPainter]
+/// method to obtain a [BoxPainter]. [Decoration] objects can be
+/// shared between boxes; [BoxPainter] objects can cache resources to
+/// make painting on a particular surface faster.
 abstract class Decoration {
+  /// Abstract const constructor.
   const Decoration();
+
+  /// In checked mode, throws an exception if the object is not in a
+  /// valid configuration. Otherwise, returns true.
+  ///
+  /// This is intended to be used as follows:
+  /// ```dart
+  ///   assert(myDecoration.debugAssertValid());
+  /// ```
   bool debugAssertValid() => true;
```

Comment:
Because it does something useful rather than returning something useful? I dunno. I think we have similar patterns elsewhere where we use a getter.